src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/TdsParser.cs

```diff
@@ -674,8 +684,13 @@ private void SendPreLoginHandshake(byte[] instanceName, bool encrypt)
                     case (int)PreLoginOptions.ENCRYPT:
                         if (_encryptionOption == EncryptionOptions.NOT_SUP)
```

Comment:
This if condition could be avoided if error is handled for this usecase beforehand.